pkg/build/strategies/strategies.go

```diff
@@ -1,23 +1,51 @@
 package strategies
 
 import (
-	"strings"
-
 	"github.com/openshift/source-to-image/pkg/api"
 	"github.com/openshift/source-to-image/pkg/build"
 	"github.com/openshift/source-to-image/pkg/build/strategies/onbuild"
 	"github.com/openshift/source-to-image/pkg/build/strategies/sti"
+	"github.com/openshift/source-to-image/pkg/docker"
 )
 
 // GetStrategy decides what build strategy will be used for the STI build.
 func GetStrategy(request *api.Request) (build.Builder, error) {
-	// TODO: The ONBUILD instructions should be available in ImageInspect call,
-	// however they are not in the current go-dockerclient version.
-	// Right now we assume that if the base image has '-onbuild' in the tag, then
-	// we do OnBuild build.
-	if strings.Contains(request.BaseImage, "-onbuild") {
+	image, err := GetBaseImage(request)
+	if err != nil {
+		return nil, err
+	}
+
+	if image.OnBuild {
 		return onbuild.New(request)
 	}
 
 	return sti.New(request)
 }
+
+// GetBaseImage processes the request and performs operations necessary to make
+// the Docker image specified as BaseImage available locally.
+// It returns informations about the base image, containing metadata necessary
+// for choosing the right STI build strategy.
+func GetBaseImage(request *api.Request) (*docker.PullResult, error) {
+	d, err := docker.New(request.DockerSocket)
+	result := docker.PullResult{}
+	if err != nil {
+		return nil, err
+	}
+	if !request.ForcePull {
+		image, err := d.CheckAndPull(request.BaseImage)
+		if err != nil {
+			return nil, err
+		}
+		result.Image = image
+		result.OnBuild = d.IsImageOnBuild(request.BaseImage)
+		return &result, nil
+	}
+	image, err := d.PullImage(request.BaseImage)
+	if err != nil {
+		return nil, err
+	}
+	result.Image = image
+	result.OnBuild = d.IsImageOnBuild(request.BaseImage)
+	return &result, nil
+}
```

Comment:
yeah, I'm gettin lazy these days ;-) I must done some refactor in docker.go and I forgot to make this func nice ;-)